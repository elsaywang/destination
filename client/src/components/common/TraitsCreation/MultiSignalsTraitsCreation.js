import React from 'react';
import PropTypes from 'prop-types';
import styles from './TraitsCreation.css';
import Button from '@react/react-spectrum/Button';
import Add from '@react/react-spectrum/Icon/Add';

const MultiSignalsTraitsCreation = ({
    selectedSignals,
    storeSessionAndNavigateToTraits,
    isMaxSignalSelectionsReached,
}) => {
    const {
        selectionMessage,
        hasSignalSelectionsTypeWarning,
        hasOnboardedSignalSelectionsWarning,
    } = selectedSignals;

    const isTraitCreationDisabled =
        hasSignalSelectionsTypeWarning ||
        isMaxSignalSelectionsReached ||
        hasOnboardedSignalSelectionsWarning;

    return (
        <div className={styles.multiCreation}>
            <span className={styles.message}>{selectionMessage}</span>
            <Button
                onClick={storeSessionAndNavigateToTraits}
                label="Create Trait From Multiple Signals"
                icon={<Add />}
                variant="action"
                disabled={isTraitCreationDisabled}
                data-test="multi-signals-trait-creation"
            />
        </div>
    );
};

MultiSignalsTraitsCreation.propTypes = {
    selectedSignals: PropTypes.shape({
        selectionMessage: PropTypes.string,
        hasSignalSelectionsTypeWarning: PropTypes.bool,
        hasOnboardedSignalSelectionsWarning: PropTypes.bool,
        selectedRowIndexes: PropTypes.array,
    }),
    storeSessionAndNavigateToTraits: PropTypes.func.isRequired,
    isMaxSignalSelectionsReached: PropTypes.bool,
};

export default MultiSignalsTraitsCreation;
