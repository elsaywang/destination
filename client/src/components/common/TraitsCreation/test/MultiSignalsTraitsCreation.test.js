import React from 'react';
import { shallow } from 'enzyme';
import MultiSignalsTraitsCreation from '../MultiSignalsTraitsCreation';
import Button from '@react/react-spectrum/Button';
import Add from '@react/react-spectrum/Icon/Add';

describe('<MultiSignalsTraitsCreation/> component', () => {
    const mockFn = jest.fn();
    const props = {
        selectedSignals: { selectionMessage: '', hasWarning: false },
        traitsCreationLabelText: 'Create Trait From Multi Signals',
        handleTraitsCreation: mockFn,
    };
    const wrapper = shallow(<MultiSignalsTraitsCreation {...props} />);

    it('matches the snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders the correct message passed from `selectedSignals.selectionMessage` prop in a <span/>', () => {
        expect(
            wrapper.containsMatchingElement(<span>{props.selectedSignals.selectionMessage}</span>),
        ).toBeTruthy();
    });

    it('renders <Button /> component has correct labeling passed from `traitsCreationLabelText` prop', () => {
        expect(wrapper.find(Button).exists()).toBeTruthy();
        expect(wrapper.find(Button).props().label).toEqual(props.traitsCreationLabelText);
    });

    it('renders <Button /> component includes <Add/> as an icon prop', () => {
        expect(wrapper.find(Button).props().icon).toEqual(<Add />);
    });

    it('renders non-disabled <Button /> with "action" variant when no warning message passes from props', () => {
        expect(wrapper.find(Button).props().variant).toEqual('action');
        expect('disabled' in wrapper.props()).toBeFalsy();
    });

    it('renders disabled <Button /> with "action" variant when warning message passes from props', () => {
        const propsWithWarning = {
            selectedSignals: {
                selectionMessage: '1 Real-time signal , 1 Onboarded signal selected ',
                records: [
                    { rowIndex: 0, signalType: 'Adobe Analytics' },
                    { rowIndex: 1, signalType: 'Onboarded Records' },
                ],
                hasWarning: true,
            },
        };
        wrapper.setProps({ ...propsWithWarning });
        expect(wrapper.find(Button).props().variant).toEqual('action');
        expect(wrapper.find(Button).props().disabled).toBeTruthy();
    });
});
