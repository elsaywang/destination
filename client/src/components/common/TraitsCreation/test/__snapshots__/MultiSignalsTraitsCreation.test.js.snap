// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MultiSignalsTraitsCreation/> component matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MultiSignalsTraitsCreation
    isMaxSignalSelectionsReached={false}
    selectedSignals={
        Object {
            "hasSignalSelectionsTypeWarning": false,
            "selectionMessage": "",
          }
    }
    storeSessionAndNavigateToTraits={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="message"
/>,
        <Button
          autoFocus={false}
          block={false}
          data-test="multi-signals-trait-creation"
          disabled={false}
          element="button"
          icon={<Add />}
          invalid={false}
          label="Create Trait From Multiple Signals"
          logic={false}
          onClick={[Function]}
          quiet={false}
          selected={false}
          variant="action"
/>,
      ],
      "className": "multiCreation",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "message",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "block": false,
          "data-test": "multi-signals-trait-creation",
          "disabled": false,
          "element": "button",
          "icon": <Add />,
          "invalid": false,
          "label": "Create Trait From Multiple Signals",
          "logic": false,
          "onClick": [Function],
          "quiet": false,
          "selected": false,
          "variant": "action",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="message"
/>,
          <Button
            autoFocus={false}
            block={false}
            data-test="multi-signals-trait-creation"
            disabled={false}
            element="button"
            icon={<Add />}
            invalid={false}
            label="Create Trait From Multiple Signals"
            logic={false}
            onClick={[Function]}
            quiet={false}
            selected={false}
            variant="action"
/>,
        ],
        "className": "multiCreation",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "message",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "block": false,
            "data-test": "multi-signals-trait-creation",
            "disabled": false,
            "element": "button",
            "icon": <Add />,
            "invalid": false,
            "label": "Create Trait From Multiple Signals",
            "logic": false,
            "onClick": [Function],
            "quiet": false,
            "selected": false,
            "variant": "action",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
